src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -92,7 +93,8 @@ private Optional<EppoValue> getAssignmentValue(
         }
 
         // Check if in experiment sample
-        Allocation allocation = configuration.getAllocation(rule.get().getAllocationKey());
+        String allocationKey = rule.get().getAllocationKey();
```

Comment:
@leoromanovsky 

Re: your question about how to log the allocation key when it is not found. Looking at the code now, it looks like we don't need to consider this scenario. Every targeting rule in the RAC has an allocation key (it's not optional in the dto): 

https://github.com/Eppo-exp/eppo/blob/127e5da05cfb391f280a7304a75b77f1d1f7624c/packages/api/src/randomized-assignment/dto/randomized-assignment-targeting-rule.dto.ts#L5

And it looks like all the SDKs I've looked at early return if a matching rule isn't found. So you see that here in the Java code with the `rule.isEmpty()` check and `return Optional.empty()`. I also see it in Python 

https://github.com/Eppo-exp/python-sdk/blob/d20b7d6d413969040ffc1246886ce2debbc5f881/eppo_client/client.py#L60-L66

and Ruby

https://github.com/Eppo-exp/ruby-sdk/blob/9de9ea4e60c509f731a97194416ccd60bee19836/lib/client.rb#L44-L50

So if we get this far into the "getAssignment" function (which we'd need to in order to log the assignment) then `allocationKey` won't ever be undefined (or at least that's how it looks to me from reading through the code).